include/openPMD/RecordComponent.hpp

```diff
@@ -127,6 +127,16 @@ class RecordComponent : public BaseRecordComponent
     template< typename T >
     RecordComponent& makeEmpty( uint8_t dimensions );
 
+    /**
+     * @brief Non-template overload of RecordComponent::makeEmpty(). 
+     * Uses the passed openPMD datatype to determine the template parameter.
+     * 
+     * @param dt The datatype of which to create an empty dataset.
+     * @param dimensions The dimensionality of the dataset.
+     * @return RecordComponent& 
```

Comment:
```suggestion
     * @return A reference to this RecordComponent.
```

Reply:
Orthogonal question for a follow-up PR about the return type:
with handles implemented, could we make the code usage safer by returning in all those places just a `RecordComponent` instead of a `RecordComponent&`? What do you think?

Reply:
The intent of returning this at all is chaining methods, right? So, things such as `recordComponent.makeEmpty(3).setUnitSI(3.0)`.
By forcing a copy there, we would also needlessly force copying the contained `shared_ptr`s, which we already do somewhat excessively at times, so I'm not really convinced.

I'm in general not a huge fan of how C++ does references and the problems in returning references is one of the major reasons (especially implicit copying upon `auto x = <reference>`), but do you see a safety issue in our case?

Reply:
Yes, the idea is to chain methods.

I think copying a shared_ptr is okay if it prevents that a user can create a dangling reference. It makes the API safer and keeps the syntactic sugar.